### PR TITLE
GEOMESA-783 Use pluggable authorizations provider for web security

### PR DESCRIPTION
**Pull request by anthonyccri**
* Created a new security module that can be used by
  multiple downstream modules for resolving AuthorizationProviders
  from SPI
* Web security now uses pluggable AuthorizationsProviders

Signed-off-by: Anthony Fox <anthony@ccri.com>

### CODE REVIEW

File: geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/AccumuloFeatureWriter.scala [hunk 1]
```diff
@@ -31,7 +31,8 @@ import org.geotools.filter.identity.FeatureIdImpl
 import org.locationtech.geomesa.core.data.AccumuloFeatureWriter._
 import org.locationtech.geomesa.core.data.tables.{AttributeTable, RecordTable, SpatioTemporalTable}
 import org.locationtech.geomesa.core.index._
-import org.locationtech.geomesa.utils.security.SecurityUtils.FEATURE_VISIBILITY
+import org.locationtech.geomesa.security.SecurityUtils
+import SecurityUtils.FEATURE_VISIBILITY
```
Comment (AndrewAnnex) [inline geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/AccumuloFeatureWriter.scala]: make this a one liner with line above

File: pom.xml [hunk 2]
```diff
@@ -904,6 +905,16 @@
...
             </dependency>
+            <dependency>
+                <groupId>org.locationtech.geomesa</groupId>
+                <artifactId>geomesa-security-accumulo1.5</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.locationtech.geomesa</groupId>
+                <artifactId>geomesa-web-security-accumulo1.5</artifactId>
+                <version>1.0.0-rc.6-SNAPSHOT</version>
```
Comment (mmatz-ccri) [inline pom.xml]: should be ${project.version}

File: geomesa-web/geomesa-web-security/src/main/scala/org/locationtech/geomesa/web/security/DataStoreSecurityProviderImpl.scala [hunk 3]
```diff
@@ -128,7 +128,7 @@ object GMSecureFeatureReader extends Logging {
 }
 
 class VisibilityFilter(ve: VisibilityEvaluator) extends Filter {
```
Comment (mmatz-ccri) [inline geomesa-web/geomesa-web-security/src/main/scala/org/locationtech/geomesa/web/security/DataStoreSecurityProviderImpl.scala]: Should this be moved to the new security module?

**Comment by mmatz-ccri** [comment]
Also, the reference to SecurityUtils in geomesa-web-security/README L34 needs to be updated.

File: geomesa-web/geomesa-web-security/src/main/scala/org/locationtech/geomesa/web/security/DataStoreSecurityProviderImpl.scala (hunk 3 above)
Comment (anthonyccri) [inline geomesa-web/geomesa-web-security/src/main/scala/org/locationtech/geomesa/web/security/DataStoreSecurityProviderImpl.scala]: Sure, I'll move it.

File: pom.xml (hunk 2 above)
Comment (anthonyccri) [inline pom.xml]: Done

**Comment by AndrewAnnex** [comment]
Committed in: b6e8d1b3d0cabd3c78a647c22f39630f5766dbac

**Comment by AndrewAnnex** [comment]
Committed in: b6e8d1b3d0cabd3c78a647c22f39630f5766dbac

**Pull request closed without merging**